### PR TITLE
Allow to pass Content-Length when uploading file

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

File: src/files.js [hunk 1]
```diff
@@ -61,14 +61,17 @@ function doUpload (cozy, data, method, path, options) {
     lastModifiedDate = new Date(lastModifiedDate)
   }
 
+  const headers = {
+    'Content-Type': contentType
+  }
+  if (contentLength) headers['Content-Length'] = String(contentLength)
+  if (checksum) headers['Content-MD5'] = checksum
+  if (lastModifiedDate) headers['Date'] = lastModifiedDate.toGMTString()
+  if (ifMatch) headers['If-Match'] = ifMatch
```
Comment (ptbrowne) [inline src/files.js]: Have you noticed a problem when a header was set to '' ?

File: test/integration/files.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const COZY_STACK_VERSION = process.env && process.env.COZY_STACK_VERSION
 const COZY_STACK_TOKEN = process.env && process.env.COZY_STACK_TOKEN
 
 describe('files API', async function () {
+  this.timeout(5000)
```
Comment (ptbrowne) [inline test/integration/files.js]: TIL

File: test/integration/files.js [hunk 3]
```diff
@@ -59,6 +60,47 @@ describe('files API', async function () {
...
+    stream.push('datastring1')
+    stream.push(null)
+
+    const created = await cozy.client.files.create(stream, {
+      name: filename,
+      contentType: 'application/json',
+      contentLength: 4
+    })
+
+    // created.should.be.rejectedWith(/content size/)
```
Comment (ptbrowne) [inline test/integration/files.js]: why is it commented ?

File: test/integration/files.js [hunk 4]
```diff
@@ -59,6 +60,47 @@ describe('files API', async function () {
...
+    stream.push('datastring1')
+    stream.push(null)
+    // stream.timeout = 2000
+
+    const oldFetch = global.fetch
+    global.fetch = (url, opts) => {
+      if (opts) opts.timeout = 500
+      return oldFetch(url, opts)
+    }
+    after(() => { global.fetch = oldFetch })
```
Comment (ptbrowne) [inline test/integration/files.js]: I did not know about this after() function and I cannot see it in the [Jest doc](https://facebook.github.io/jest/docs/en/api.html). Where does it come from ?

Comment (ptbrowne) [reply test/integration/files.js]: OK, we use mocha here.

File: test/integration/files.js (hunk 3 above)
Comment (aenario) [reply test/integration/files.js]: This was the expected behaviour but it actually ends up as a timeout

File: src/files.js (hunk 1 above)
Comment (aenario) [reply src/files.js]: I had an issue with an undefined header wich was sent as "undefined", I feel this better express the need.

File: test/integration/files.js (hunk 3 above)
Comment (aenario) [reply test/integration/files.js]: Correction : The expected behaviour was that we got an error from the stack, but somewhere in the node-fetch / http.Request mechanism, the request actually get truncated.

File: src/files.js (hunk 1 above)
Comment (ptbrowne) [reply src/files.js]: 👍

File: test/integration/files.js (hunk 3 above)
Comment (aenario) [reply test/integration/files.js]: removed commented code

**Pull request merged**